src/plugins/plot/src/configuration/Collection.js

```diff
@@ -113,13 +113,23 @@ define([
         );
     };
 
-    Collection.prototype.remove = function (model) {
-        var index = this.indexOf(model);
-        if (index === -1) {
-            throw new Error('model not found in collection.');
+    Collection.prototype.remove = function (model, identifier) {
```

Comment:
I think if we restrict the change to `SeriesCollection` (which extends Collection) by overriding `remove`, then we can safely assume we're always dealing with domain object models and always use `model.domainObject.identifier` without having to pass in a separate identifier. This will simplify the logic.

Reply:
Great call, i was able to keep the changes specifically to series collection and leave the remove function unchanged.